.github/actions/scala_build/action.yml

```diff
@@ -45,7 +45,9 @@ runs:
           run: sudo mvn -q clean install -DskipTests -Dscoverage.skip
         - name: Publish test coverage
           if: inputs.skip_tests == 'false'
-          uses: codecov/codecov-action@v1
+          uses: codecov/codecov-action@v4
```

Comment:
I would move this step to the workflows, this could be a step in build_main.yml workflow.
sectrets should be resolved correctly there